src/app/api/index.ts

```diff
@@ -308,8 +311,8 @@ class Client implements API {
         //This is a hack to create a unique id for a new assignment
         newAssignment.id = assignments.length;
         //set the assignment title
-        if (WORK_SECTIONS[newAssignment.workSectionId] === WORK_SECTIONS.COURTS) {
-            newAssignment.title = COURTROOMS[newAssignment.title];
+        if (WORK_SECTIONS[newAssignment.workSectionId] === WORK_SECTIONS.COURTS && newAssignment.courtroomId) {
```

Comment:
this looks very similar to what is in the `setAssignmentTemplateTitle` below.  Perhaps we could use that here to replace this logic.

```ts
newAssignment.title = getAssignmentTitle(newAssignment);
```